boa/src/object/gcobject.rs

```diff
@@ -992,6 +1010,112 @@ impl GcObject {
         // 7. Return target.
         Ok(())
     }
+
+    pub fn set_integrity_level(
+        &self,
+        level: IntegrityLevel,
+        context: &mut Context,
+    ) -> JsResult<bool> {
+        // 1. Assert: Type(O) is Object.
+        // 2. Assert: level is either sealed or frozen.
+
+        // 3. Let status be ? O.[[PreventExtensions]]().
+        let status = self.__prevent_extensions__();
+        // 4. If status is false, return false.
+        if !status {
+            return Ok(false);
+        }
+
+        // 5. Let keys be ? O.[[OwnPropertyKeys]]().
+        let keys = self.own_property_keys();
+
+        match level {
+            // 6. If level is sealed, then
+            IntegrityLevel::Sealed => {
+                // a. For each element k of keys, do
+                for k in keys {
+                    // i. Perform ? DefinePropertyOrThrow(O, k, PropertyDescriptor { [[Configurable]]: false }).
+                    self.define_property_or_throw(
+                        k,
+                        PropertyDescriptor::builder().configurable(false).build(),
+                        context,
+                    )?;
+                }
+            }
+            // 7. Else,
+            //     a. Assert: level is frozen.
+            IntegrityLevel::Frozen => {
+                // b. For each element k of keys, do
+                for k in keys {
+                    // i. Let currentDesc be ? O.[[GetOwnProperty]](k).
+                    let current_desc = self.__get_own_property__(&k);
+                    // ii. If currentDesc is not undefined, then
+                    if let Some(current_desc) = current_desc {
+                        // 1. If IsAccessorDescriptor(currentDesc) is true, then
+                        let desc = if current_desc.is_accessor_descriptor() {
+                            // a. Let desc be the PropertyDescriptor { [[Configurable]]: false }.
+                            PropertyDescriptor::builder().configurable(false).build()
+                        // 2. Else,
+                        } else {
+                            // a. Let desc be the PropertyDescriptor { [[Configurable]]: false, [[Writable]]: false }.
+                            PropertyDescriptor::builder()
+                                .configurable(false)
+                                .writable(false)
+                                .build()
+                        };
+                        // 3. Perform ? DefinePropertyOrThrow(O, k, desc).
+                        self.define_property_or_throw(k, desc, context)?;
+                    }
+                }
+            }
+        }
+
+        // 8. Return true.
+        Ok(true)
+    }
+
+    pub fn test_integrity_level(
```

Comment:
Another place for documentation :)

Reply:
This function also should be on `object::operations`